### PR TITLE
Implement SDL/dx11 sample

### PR DESCRIPTION
**Pull request by vincenthamm**
Implement SDL/dx11 sample
Add support for viewport when using SDL with dx11

Addressing comments mentioned in #2482

### CODE REVIEW

**Comment by ocornut** [comment]
Thank you Vincent.
What I will do is to first merge the equivalent code in the `master` branch, then in a second step merge the changes for multi-viewports. If you can I'd suggest making the PR (I think you had the commit) for `master` first.

Since you changed _dx11 to use `PlatformHandleRaw` this is presumably breaking other back-ends/examples which don't set this value.

Linking to #2612.

File: examples/imgui_impl_dx11.cpp [hunk 1]
```diff
@@ -564,8 +564,8 @@ static void ImGui_ImplDX11_CreateWindow(ImGuiViewport* viewport)
     ImGuiViewportDataDx11* data = IM_NEW(ImGuiViewportDataDx11)();
     viewport->RendererUserData = data;
 
-    HWND hwnd = (HWND)viewport->PlatformHandle;
-    IM_ASSERT(hwnd != 0);
+    assert(viewport->PlatformHandleRaw);
```
Comment (ocornut) [inline examples/imgui_impl_dx11.cpp]: IM_ASSERT

File: examples/imgui_examples.sln [hunk 2]
```diff
@@ -1,68 +1,78 @@
...
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "example_win32_directx11", "example_win32_directx11\example_win32_directx11.vcxproj", "{9F316E83-5AE5-4939-A723-305A94F48005}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "example_glfw_opengl2", "example_glfw_opengl2\example_glfw_opengl2.vcxproj", "{9CDA7840-B7A5-496D-A527-E95571496D18}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "example_glfw_opengl3", "example_glfw_opengl3\example_glfw_opengl3.vcxproj", "{4A1FB5EA-22F5-42A8-AB92-1D2DF5D47FB9}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "example_sdl_directx11.vcxproj", "example_sdl_directx11\example_sdl_directx11.vcxproj.vcxproj", "{9E1987E3-1F19-45CA-B9C9-D31E791836D8}"
+EndProject
+Global
```
Comment (ocornut) [inline examples/imgui_examples.sln]: Wouldn't add this to the default SLN as there's no reliable way to guarantee it'll compile under Windows without SDL installed somewhere in a folder accessible by the compiler.

**Comment by vincenthamm** [comment]
Sure, I'll split it up later today.
For the PlatformHandleRaw and other backend, I'll change so that we use PlatformHandle in the old way if PlatformHandleRaw is null. I think that should make other backend work without changes.

**Comment by vincenthamm** [comment]
Superseded by #2630

**Pull request closed without merging**